web/src/app/services/selected-term.service.ts

```diff
@@ -0,0 +1,106 @@
+import {Injectable} from '@angular/core';
+import {Term} from 'yacs-api-client';
+import {Subject, Subscription} from 'rxjs';
+
+@Injectable()
+export class SelectedTermService {
+    // a subject that can be written to and subscribed to for the selected term
+    public selectedTerm$: Subject<Term>;
+    // hold the terms in two forms: a mapping between the uuid and the term, and an array for ordering
+    protected terms: Map<string, Term>;
+    protected termsOrdinal: Term[];
+    // cache the current ordinal and uuid
+    protected _currentOrdinal: number;
+    protected _currentUUID: string;
+    protected _currentId: string;
+
+    constructor() {
+        this.terms = new Map<string, Term>();
+        this.termsOrdinal = new Array<Term>();
```

Comment:
From what I can tell although a `Map` is ordered by insertion, indexing into it is trickier; it will require an `Array.from(...)` or similar spread operator use to directly index a given value by ordinal. Alternatively, I could loop through calling `Iterator.next()` `n` times. (Unless you know a better way to directly get a Map entry by insertion index >.>)

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/values

Reply:
also https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/keys and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/entries

these all return stuff in insertion order